master/include/logic/stateLogic.hpp

```diff
@@ -1,104 +1,108 @@
-    };
+    explicit ASState(SystemData *system_data, Communicator *communicator) : _checkupManager(system_data),
+                                                                            _communicator(communicator) {
+    }
 
     void calculateState();
+
+    void performEmergencyOperations();
 };
 
 inline void ASState::calculateState() {
     switch (state) {
         case AS_MANUAL:
-            if (!_checkupManager.manualDrivingCheckup())
-                break;
+            if (_checkupManager.manualDrivingCheckup()) break;
 
-        // TODO: EBS from disabled to inactive
-        // TODO: Open SDC circuit
+            DigitalSender::enterOffState();
 
             state = AS_OFF;
             break;
 
         case AS_OFF:
+
             // If manual driving checkup fails, the car can't be in OFF state, so it goes back to MANUAL
             if (_checkupManager.manualDrivingCheckup()) {
-                //TODO: OFF -> MANUAL Operations
+                DigitalSender::enterManualState();
                 state = AS_MANUAL;
                 break;
             }
-            if (_checkupManager.offCheckup())
-                break;
 
-            // TODO: OFF -> READY Operations
+            if (_checkupManager.offCheckup()) break;
+
+            DigitalSender::enterReadyState();
             state = AS_READY;
             break;
 
         case AS_READY:
-            if (_checkupManager.emergencyCheckup()) {
-                // Emergency, go to emergency state
-                /* TODO:
-                ASSI to blue flashing
-                EBS to Active
-                TS to OFF (open sdc)
-                Buzzer to on for 8-10s
-                */
-                state = AS_EMERGENCY;
+            if (_checkupManager.readyCheckup()) {
+                DigitalSender::enterOffState();
+                state = AS_OFF;
                 break;
             }
-            if (_checkupManager.r2dCheckup()) {
-                // Not ready, do nothing
+
+            if (_checkupManager.emergencyCheckup()) {
+                performEmergencyOperations();
+                state = AS_EMERGENCY;
                 break;
             }
-        /* TODO:
-        Ready to drive, go to driving state
-        ASSI to yellow flashing
-        EBS to inactive
-        */
+            if (_checkupManager.r2dCheckup()) break;
+
+            _digitalSender.enterDrivingState();
             state = AS_DRIVING;
             break;
         case AS_DRIVING:
+            _digitalSender.blinkLED();
             if (_checkupManager.emergencyCheckup()) {
-                // Emergency, go to emergency state
+                performEmergencyOperations();
                 state = AS_EMERGENCY;
                 break;
             }
-            if (_checkupManager.missionFinishedCheckup()) {
-                break;
-            }
+            if (_checkupManager.drivingCheckup()) break;
```

Comment:
I think the combination of having a successfull exit as a 0 (which is typical in some contexts but may be unexpected in this one, as usually it is coupled with an error check) and the fact that you use the break instead of reversing the logic and puting the rest inside of the if statement leaves this code more confusing than it should be.